### PR TITLE
meta: upgrade dependencies

### PR DESCRIPTION
**Pull request by Harshit933**
LDK 0.0.123 -> 0.0.125
bitcoin 0.30 -> 0.32 (as LDK uses 0.32)
bitcoincore_rpc 0.17 -> 0.19

### CODE REVIEW

**Comment by Harshit933** [comment]
This is work in progress

```
error[E0277]: the trait bound `HexIterator<'_>: bitcoincore_rpc::bitcoin::bitcoin_io::Read` is not satisfied
   --> lampo-core-wallet/src/lib.rs:276:50
    |
276 |         let object = Decodable::consensus_decode(&mut reader)?;
    |                      --------------------------- ^^^^^^^^^^^ the trait `bitcoincore_rpc::bitcoin::bitcoin_io::Read` is not implemented for `HexIterator<'_>`
    |                      |
    |                      required by a bound introduced by this call
    |
    = help: the following other types implement trait `bitcoincore_rpc::bitcoin::bitcoin_io::Read`:
              &File
              &Stdin
              &TcpStream
              &UnixStream
              &[u8]
              Arc<File>
              BufReader<R>
              ChildStderr
            and 19 others
note: required by a bound in `bitcoincore_rpc::bitcoin::consensus::Decodable::consensus_decode`
   --> /home/harshit/.cargo/registry/src/index.crates.io-6f17d22bba15001f/bitcoin-0.32.4/src/consensus/encode.rs:371:28
    |
371 |     fn consensus_decode<R: Read + ?Sized>(reader: &mut R) -> Result<Self, Error> {
    |                            ^^^^ required by this bound in `Decodable::consensus_decode`

warning: unreachable pattern
  --> lampo-core-wallet/src/lib.rs:87:13
   |
87 |             _ => unreachable!(),
   |             ^
   |
   = note: `#[warn(unreachable_patterns)]` on by default

warning: use of deprecated method `lampo_common::backend::Transaction::txid`: txid has been renamed to compute_txid to note that it's computationally expensive.  use compute_txid() instead.
   --> lampod/src/actions/handler.rs:195:76
    |
195 |                 log::info!("funding transaction created `{}`", transaction.txid());
    |                                                                            ^^^^
    |
    = note: `#[warn(deprecated)]` on by default

warning: use of deprecated method `lampo_common::backend::Transaction::txid`: txid has been renamed to compute_txid to note that it's computationally expensive.  use compute_txid() instead.
   --> lampod/src/ln/channel_manager.rs:151:127
    |
151 |                         log::info!(target: "channel_manager", "confirmed transaction with txid `{}` at height `{height}`", tx.txid());
    |                                                                                                                               ^^^^

For more information about this error, try `rustc --explain E0277`.
warning: `lampo-core-wallet` (lib) generated 1 warning
error: could not compile `lampo-core-wallet` (lib) due to 1 previous error; 1 warning emitted
warning: build failed, waiting for other jobs to finish...
warning: use of deprecated method `lampo_common::backend::Transaction::txid`: txid has been renamed to compute_txid to note that it's computationally expensive.  use compute_txid() instead.
   --> lampod/src/ln/channel_manager.rs:435:44
    |
435 |         let txid = tx.as_ref().map(|tx| tx.txid());
    |                                            ^^^^

error[E0277]: the `?` operator can only be used on `Result`s in a method that returns `Result`
  --> lampod/src/ln/offchain_manager.rs:88:87
   |
85 |     pub fn decode_invoice(&self, invoice_str: &str) -> error::Result<ldk::invoice::Bolt11Invoice> {
   |     --------------------------------------------------------------------------------------------- this function returns a `Result`
...
88 |             .map_err(|err| error::bail!("Error occured while decoding invoice {err}"))?;
   |                                                                                       ^ this `?` produces `Result<Infallible, Result<_, lampo_common::error::Error>>`, which is incompatible with `Result<Bolt11Invoice, lampo_common::error::Error>`
   |
   = help: the trait `StdError` is not implemented for `Result<Bolt11Invoice, lampo_common::error::Error>`, which is required by `Result<Bolt11Invoice, lampo_common::error::Error>: FromResidual<Result<Infallible, Result<_, lampo_common::error::Error>>>`
   = help: the following other types implement trait `FromResidual<R>`:
             <Result<T, F> as FromResidual<Result<Infallible, E>>>
             <Result<T, F> as FromResidual<Yeet<E>>>
   = note: required for `lampo_common::error::Error` to implement `From<Result<_, lampo_common::error::Error>>`
   = note: required for `Result<Bolt11Invoice, lampo_common::error::Error>` to implement `FromResidual<Result<Infallible, Result<_, lampo_common::error::Error>>>`

error[E0271]: expected `{closure@lib.rs:253:29}` to be a closure that returns `Result<(), ReplayEvent>`, but it returns `()`
   --> lampod/src/lib.rs:269:13
    |
267 |         let background_processor = BackgroundProcessor::start(
    |                                    -------------------------- required by a bound introduced by this call
268 |             self.persister.clone(),
269 |             event_handler,
    |             ^^^^^^^^^^^^^ expected `Result<(), ReplayEvent>`, found `()`
    |
    = note:   expected enum `Result<(), ReplayEvent>`
            found unit type `()`
    = note: required for `{closure@lampod/src/lib.rs:253:29: 253:48}` to implement `lampo_common::ldk::events::EventHandler`
note: required by a bound in `BackgroundProcessor::start`
   --> /home/harshit/.cargo/registry/src/index.crates.io-6f17d22bba15001f/lightning-background-processor-0.0.125/src/lib.rs:913:17
    |
904 |     pub fn start<
    |            ----- required by a bound in this associated function
...
913 |         EH: 'static + EventHandler + Send,
    |                       ^^^^^^^^^^^^ required by this bound in `BackgroundProcessor::start`

error[E0277]: the trait bound `LampoLogger: APeerManager` is not satisfied
   --> lampod/src/lib.rs:274:13
    |
267 |         let background_processor = BackgroundProcessor::start(
    |                                    -------------------------- required by a bound introduced by this call
...
274 |             self.logger.clone(),
    |             ^^^^^^^^^^^^^^^^^^^ the trait `APeerManager` is not implemented for `LampoLogger`
    |
    = help: the trait `APeerManager` is implemented for `PeerManager<Descriptor, CM, RM, OM, L, CMH, NS>`
note: required by a bound in `BackgroundProcessor::start`
   --> /home/harshit/.cargo/registry/src/index.crates.io-6f17d22bba15001f/lightning-background-processor-0.0.125/src/lib.rs:941:15
    |
904 |     pub fn start<
    |            ----- required by a bound in this associated function
...
941 |         PM::Target: APeerManager + Send + Sync,
    |                     ^^^^^^^^^^^^ required by this bound in `BackgroundProcessor::start`

error[E0061]: this function takes 9 arguments but 8 arguments were supplied
   --> lampod/src/lib.rs:267:36
    |
267 |           let background_processor = BackgroundProcessor::start(
    |  ____________________________________^^^^^^^^^^^^^^^^^^^^^^^^^^-
268 | |             self.persister.clone(),
269 | |             event_handler,
270 | |             self.channel_manager().chain_monitor(),
271 | |             self.channel_manager().manager(),
272 | |             GossipSync::p2p(gossip_sync),
    | |             ---------------------------- expected `std::option::Option<_>`, found `GossipSync<Arc<P2PGossipSync<Arc<NetworkGraph<Arc<LampoLogger>>>, Arc<LampoChainManager>, Arc<LampoLogger>>>, &lightning_rapid_gossip_sync::RapidGossipSync<Arc<NetworkGraph<Arc<LampoLogger>>>, Arc<LampoLogger>>, Arc<NetworkGraph<Arc<LampoLogger>>>, Arc<LampoChainManager>, Arc<LampoLogger>>`
273 | |             self.peer_manager().manager(),
    | |             ----------------------------- unexpected argument of type `Arc<PeerManager<lampo_common::ldk::lightning_net_tokio::SocketDescriptor, Arc<ChannelManager<Arc<ChainMonitor<InMemorySigner, Arc<(dyn lampo_common::ldk::chain::Filter + std::marker::Send + std::marker::Sync + 'static)>, Arc<LampoChainManager>, Arc<LampoChainManager>, Arc<LampoLogger>, Arc<FilesystemStore>>>, Arc<LampoChainManager>, Arc<LampoKeysManager>, Arc<LampoKeysManager>, Arc<LampoKeysManager>, Arc<LampoChainManager>, Arc<DefaultRouter<Arc<NetworkGraph<Arc<LampoLogger>>>, Arc<LampoLogger>, Arc<LampoKeysManager>, Arc<std::sync::Mutex<ProbabilisticScorer<Arc<NetworkGraph<Arc<LampoLogger>>>, Arc<LampoLogger>>>>, ProbabilisticScoringFeeParameters, ProbabilisticScorer<Arc<NetworkGraph<Arc<LampoLogger>>>, Arc<LampoLogger>>>>, Arc<LampoLogger>>>, Arc<P2PGossipSync<Arc<NetworkGraph<Arc<LampoLogger>>>, Arc<LampoChainManager>, Arc<LampoLogger>>>, Arc<OnionMessenger<Arc<LampoKeysManager>, Arc<LampoKeysManager>, Arc<LampoLogger>, Arc<EmptyNodeIdLookUp>, Arc<DefaultMessageRouter<Arc<NetworkGraph<Arc<LampoLogger>>>, Arc<LampoLogger>, Arc<LampoKeysManager>>>, IgnoringMessageHandler, IgnoringMessageHandler, IgnoringMessageHandler>>, Arc<LampoLogger>, IgnoringMessageHandler, Arc<LampoKeysManager>>>`
274 | |             self.logger.clone(),
275 | |             Some(self.channel_manager().scorer()),
    | |             ------------------------------------- expected `GossipSync<_, _, _, _, Arc<LampoLogger>>`, found `std::option::Option<Arc<std::sync::Mutex<ProbabilisticScorer<Arc<NetworkGraph<Arc<LampoLogger>>>, Arc<LampoLogger>>>>>`
276 | |         );
    | |_________- two arguments of type `Arc<LampoLogger>` and `std::option::Option<_>` are missing
    |
note: associated function defined here
   --> /home/harshit/.cargo/registry/src/index.crates.io-6f17d22bba15001f/lightning-background-processor-0.0.125/src/lib.rs:904:9
    |
904 |     pub fn start<
    |            ^^^^^
help: did you mean
    |
267 |         let background_processor = BackgroundProcessor::start(self.persister.clone(), event_handler, self.channel_manager().chain_monitor(), self.channel_manager().manager(), Some(self.channel_manager().scorer()), GossipSync::p2p(gossip_sync), self.logger.clone(), /* Arc<LampoLogger> */, /* scorer */);
    |                                                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Some errors have detailed explanations: E0061, E0271, E0277.
For more information about an error, try `rustc --explain E0061`.
warning: `lampod` (lib) generated 5 warnings
error: could not compile `lampod` (lib) due to 4 previous errors; 5 warnings emitted
```

**Comment by vincenzopalazzo** [comment]
When you are done with this, can you do an update package per commit?

It is a blob change anyway, however, if you wait https://github.com/vincenzopalazzo/lampo.rs/pull/292 this should be way simple because there are less packages that depends from `rust-bitcoin`

**Comment by Harshit933** [comment]
Last steps https://github.com/rust-bitcoin/rust-bitcoin/issues/3668

**Comment by Harshit933** [comment]
Some tests are failing

```
test lampo_tests::init_connection_test_between_lampo ... ok
test lampo_cln_tests::no_able_to_pay_invoice_to_cln ... ok
test lampo_cln_tests::init_connection_test ... ok
test lampo_cln_tests::decode_invoice_from_cln ... ok
test lampo_tests::decode_offer ... FAILED
test lampo_cln_tests::test_close_channel_without_opening_a_channel_fails ... ok
test lampo_tests::pay_invoice_simple_case_lampo ... FAILED
test lampo_tests::pay_offer_minimal_offer ... FAILED
test lampo_cln_tests::decode_cln_offer_from_lampo ... FAILED
test lampo_cln_tests::decode_cln_offer_from_lampo_minimal_offer ... FAILED
test lampo_cln_tests::fund_a_simple_channel_to_lampo ... ok
test lampo_tests::fund_a_simple_channel_from ... FAILED
test lampo_cln_tests::test_lampo_to_cln_close_channel_with_channel_id_success ... FAILED
test lampo_cln_tests::payinvoice_to_lampo ... FAILED
test lampo_cln_tests::test_lampo_to_cln_close_channel_without_channel_id_success ... FAILED
test lampo_cln_tests::pay_invoice_to_cln ... FAILED
test lampo_cln_tests::test_closing_two_channels_without_channelid_fails ... FAILED
test lampo_cln_tests::fund_a_simple_channel_from_lampo ... FAILED
test lampo_tests::pay_offer_simple_case_lampo ... FAILED
test lampo_cln_tests::test_sync_gossip_from_network_cln ... ok
test lampo_cln_tests::be_able_to_kesend_payments ... FAILED
```

**Comment by Harshit933** [comment]
Fixed the tests

File: lampo-common/src/keys.rs [hunk 1]
```diff
@@ -1,8 +1,10 @@
 use std::{sync::Arc, time::SystemTime};
 
-use bitcoin::secp256k1::{Secp256k1, SecretKey};
 use lightning::sign::{InMemorySigner, NodeSigner, OutputSpender, SignerProvider};
+use lightning_invoice::RawBolt11Invoice;
 
+use crate::ldk::bitcoin::secp256k1::Secp256k1;
+use crate::ldk::bitcoin::secp256k1::SecretKey;
```
Comment (vincenzopalazzo) [inline lampo-common/src/keys.rs:7]: we should have access to `use crate::bitcoin`

File: lampo-common/src/keys.rs [hunk 2]
```diff
@@ -113,9 +115,9 @@ impl NodeSigner for LampoKeysManager {
     fn ecdh(
         &self,
         recipient: lightning::sign::Recipient,
-        other_key: &bitcoin::secp256k1::PublicKey,
-        tweak: Option<&bitcoin::secp256k1::Scalar>,
-    ) -> Result<bitcoin::secp256k1::ecdh::SharedSecret, ()> {
+        other_key: &lightning::bitcoin::secp256k1::PublicKey,
+        tweak: Option<&lightning::bitcoin::secp256k1::Scalar>,
+    ) -> Result<lightning::bitcoin::secp256k1::ecdh::SharedSecret, ()> {
```
Comment (vincenzopalazzo) [inline lampo-common/src/keys.rs:120]: same here we do not referer in this crate with `lightning`, it is wrong.

Please see how we use it in other part of the code

File: lampo-common/src/model.rs [hunk 3]
```diff
@@ -17,6 +17,7 @@ pub mod request {
     pub use crate::model::getinfo::*;
     pub use crate::model::invoice::request::*;
     pub use crate::model::keysend::request::*;
+    #[allow(unused_imports)]
```
Comment (vincenzopalazzo) [inline lampo-common/src/model.rs:20]: I think this is wrong, why we are using it? you are suppressing warning and we should not do without a good reason

File: lampo-common/src/model/invoice.rs [hunk 4]
```diff
@@ -42,11 +41,11 @@ pub mod response {
         pub bolt11: String,
     }
 
-    #[derive(Serialize, Deserialize, Debug)]
+    #[derive(Debug, Serialize, Deserialize)]
```
Comment (vincenzopalazzo) [inline lampo-common/src/model/invoice.rs:44]: Why this unnecessary change?

**Review by vincenzopalazzo — CHANGES_REQUESTED** [review]
Thanks for this.

I am added some basic comment, and I think this PR should be based in on the `next` branch because it is much cleaner.

Or you can build on top of the first comment of the following PR https://github.com/vincenzopalazzo/lampo.rs/pull/292/commits (ONLY the first commit, so you can cherry-pick it)

File: lampo-common/src/model.rs (hunk 3 above)
Comment (Harshit933) [reply lampo-common/src/model.rs:20]: The network request has nothing in it. 
https://github.com/vincenzopalazzo/lampo.rs/blob/c10138add774f428b278224a4f3b19b4d5b24ff5/lampo-common/src/model/network.rs#L1

We could probably remove this, but then the request and response pattern would be non-uniform.

File: lampo-common/src/model/invoice.rs (hunk 4 above)
Comment (Harshit933) [reply lampo-common/src/model/invoice.rs:44]: Thanks for catching

File: lampo-common/src/keys.rs (hunk 2 above)
Comment (Harshit933) [reply lampo-common/src/keys.rs:120]: Fixed

File: lampo-common/src/keys.rs (hunk 1 above)
Comment (Harshit933) [reply lampo-common/src/keys.rs:7]: Thanks, fixed

**Comment by Harshit933** [comment]
Opened in https://github.com/vincenzopalazzo/lampo.rs/pull/314
